bottlerocket-settings-models/settings-extensions/container-runtime-plugins/src/de.rs

```diff
@@ -0,0 +1,209 @@
+use byte_unit::Byte;
+use serde::{self, Deserialize, Deserializer};
+
+/// This function deserializes a size string (e.g., "64mb") or integer into an i64 value.
+///
+/// Valid formats:
+/// - Integer values (e.g., "1024" or 1024)
+/// - Size strings with units (e.g., "64mb", "1.5gb", ".5gb")
+///
+/// Invalid formats:
+/// - Negative values other than -1
+/// - Strings with invalid formats
+/// - Strings with whitespace
+pub fn deserialize_chunk_size<'de, D>(deserializer: D) -> Result<Option<i64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum StringOrIntOrNone {
+        String(String),
+        Int(i64),
+        None,
+    }
+
+    let value = StringOrIntOrNone::deserialize(deserializer)?;
+
+    match value {
+        StringOrIntOrNone::String(s) => {
+            let s = s.trim();
+
+            // Try parsing as a plain integer (bytes)
+            if let Ok(bytes) = s.parse::<i64>() {
+                if bytes < -1 {
+                    return Err(serde::de::Error::custom(format!(
+                        "Size cannot be less than -1: {s}"
+                    )));
+                }
+                return Ok(Some(bytes));
+            }
+
+            if s.contains(char::is_whitespace) {
+                return Err(serde::de::Error::custom(format!(
+                    "Size string cannot contain whitespace: '{s}'"
+                )));
+            }
+
+            // Handle the special case of ".5gb" format (byte_unit expects "0.5gb")
+            let normalized_s = if s.starts_with('.') {
+                format!("0{s}")
+            } else {
+                s.to_string()
+            };
+
+            // Try parsing as a size string with units using byte_unit, ignoring case.
+            match Byte::parse_str(&normalized_s, true) {
+                Ok(byte) => {
+                    // Convert to i64, ensuring we don't overflow
+                    let bytes = byte.as_u64();
+                    Ok(Some(bytes as i64))
```

Comment:
This is an unchecked conversion that might overflow. Better to use `i64::try_from`.